### PR TITLE
functional API intermediate output doc in faq

### LINKED ISSUE #2664: Output from intermediate layers with functional API
In the [FAQ section](http://keras.io/getting-started/faq/#how-can-i-visualize-the-output-of-an-intermediate-layer) there is an example about how to get output from intermediate layers using the Sequential API.

How is it possible to achieve that with the functional API? It is written:
> Another more flexible way of getting output from intermediate layers is to use the functional API.

Could you please give a little bit more details?
Thank you

### PR DESCRIPTION
**Pull request by kylemcdonald**
This closes #2664 where @GelliFrancesco asks for more info about what was meant by "just use the functional API".

I would show an example of getting a decoder too, but I don't see a simple way to do that without doing something like:

```python
encoded_input = Input(shape=(32,)
third = Dense(512)(merge(second, encoded_input))
```

Otherwise I get `UserWarning: Model inputs must come from a Keras Input layer, they cannot be the output of a previous non-Input layer.`

### CODE REVIEW

File: docs/templates/getting-started/faq.md [hunk 1]
```diff
@@ -146,7 +146,25 @@ layer_output = get_3rd_layer_output([X, 0])[0]
...
+```python
+inputs = Input(shape=(784,))
+first = Dense(512)(inputs)
+second = Dense(32)(first)
+third = Dense(512)(second)
+outputs = Dense(784)(third)
+model = Model(input=inputs, output=outputs)
+model.compile(optimizer='rmsprop', loss='mse')
+model.fit(X, X, ...)
+```
```
Comment (fchollet) [inline docs/templates/getting-started/faq.md]: Why not something even simpler:

```python
inputs = Input(shape=(784,))
encoded = Dense(32, activation='relu')(inputs)
decoded = Dense(784)(encoded)
model = Model(input=inputs, output=decoded)

encoder = Model(input=inputs, output=encoded)
```
Btw lack of activation function means that the stack of layers acts as a single linear transformation, which is probably not what you want... Also for an autoencoder on MNIST it would be better to use a `sigmoid` activation on the decoder layer and a `binary_crossentropy` loss, but that's a nitpick.

Even though it's a toy example... one thing I've learned is that people will actually copy and try to use toy examples from the docs that were meant purely for illustrating something, so they should be written carefully.

Comment (kylemcdonald) [inline docs/templates/getting-started/faq.md]: all around great points. changed and squashed commits. i tested with a relu encoder but removed it without realizing how important it was. and the mse was from working with a bunch of images, i always forget that mnist isn't usually treated as 'images'... thanks for the feedback!

**Comment by fchollet** [comment]
Cool, LGTM
